src/components/Filters/FilterRow/FilterRow.tsx

```diff
@@ -78,6 +80,44 @@ const getOperatorOptions = curry((operatorValue) =>
 
 const getDataPointOptions = map(normalizeOptions);
 
+const renderComponent = (Component, value, onChange) => {
+  if (!Component) {
```

Comment:
maybe would be good to check if the `Component` is present and is valid React Element via `React.isValidElement`
```js
if (isUndefined(Component && !React.isValidElement(Component))  return null;

Reply:
Good tip with `React.isValidElement`, updated

Reply:
the condition looks strange (I wrote it wrongly and you copy/pasted) 😆 
```js
if (isUndefined(Component) || !React.isValidElement(Component)) return null
```

Reply:
@ajkl2533 `Component` is here wrapped into argumentless function therefore `React.isValidElement` doesn't apply. I need to wrap the functional Component when using `useState` in `useFilterRow` hook, please see my last commit, otherwise Component is invoked and fails. A bit strange error. At least a believe it is the explanation https://dev.to/sarioglu/how-to-store-a-function-using-react-usestate-4ffh

Reply:
I simplified the hook and omitted storing of component itself in local state, still `isValidElement` checks reference to component as invalid
```
React.isValidElement(Component) -> false
React.isValidElement(<Component />) -> true
```